### PR TITLE
Proposed refactor of kubo install section in Install

### PR DESCRIPTION
**Pull request by ElPaisano**
The following PR proposes a refactor of https://docs.ipfs.tech/install/command-line/

**Latest staging link** 
To view the most up-to-date staging link, scroll to the _All checks have passed_ section at the bottom of this PR, click _Details_ next to `fleek/build`, and navigate to _Install_ -> _Kubo on the command line_

**Overview:**
- Breaks OS specific binary install steps out of linear sections and places them in tabs. This is mainly for readability / not making users scroll through steps that aren't necessary for them 
- Adds OpenBSD and FreeBSD install instructions per https://github.com/ipfs/ipfs-docs/issues/1326 - this was another reason to add tabs, as there are now 5 different binary install instructions, which would really crowd the page if we still formatted this as it currently is
- Emphasizes that the user is installing Kubo, the Go implementation of IPFS, not IPFS or another implementation. We've received feedback that this has been confusing users and readers, especially newer folks in programs like Launchpad
- Renamed title to emphasize that the user is installing Kubo
- Rewrites some content and adds stuff in so that we are more closely aligned with https://github.com/ipfs/kubo/blob/v0.17.0/README.md
  - Adds overview info similar to https://github.com/ipfs/kubo/blob/v0.17.0/README.md#what-is-kubo
  - Updates system requirements section to emphasize requirements, and more or less match https://github.com/ipfs/kubo/blob/v0.17.0/README.md#system-requirements
- ~~Add a TOC to make navigation easier~~ removed extra TOC per https://github.com/ipfs/ipfs-docs/pull/1342#issuecomment-1335351379
- Renamed the "Compile manually" subsection to "Build Kubo from source", and linked to appropriate subsection of repo
- Fixed a link and some markdown formatting 
- Removed table with pictures of OSes. We can add this back in if folks really want it back. But was just thinking a plain TOC would do the job
- Added a table for the binaries

Review requested:
- What do we think of the new title? Goal is to make it clear that this Kubo. May not be the best title tho.
- Do we like the attempt to emphasize that this is Kubo? If not, how can we reword things?
- Do we like the overview section?
- What do we think of the tabs section?
- What should be do with the https://docs.ipfs.tech/install/command-line/#which-node-should-you-use-with-the-command-line section? As a newer user to kubo, this section still confuses me. What is it's purpose? Should it be in the install section, or should it live elsewhere? SME feedback would be helpful here.
- Are the OpenBSD and FreeBSD install instructions correct?
- Are the system requirements correct?

### CODE REVIEW

**Comment by ElPaisano** [comment]
It appears to be the case that tabs in Vuepress don't play nicely with local links i.e. linking to a tab with a name `tab-name` via `#tab-name`. I noticed this when using the TOC. Seems like you sometime have to click the TOC entry for Mac and Linux multiple times for the page to scroll down, but it works fine for the other binaries. IDK why. If anyone can replicate this, LMK. Also, if I'm not imagining things, if anyone has a fix, that would be great

**Comment by ElPaisano** [comment]
Random thought:

looking at https://docs.ipfs.tech/install/ipfs-updater/#install-ipfs, might make sense to consolidate this into the install guide, since it's another way to install Kubo. Or even the whole https://docs.ipfs.tech/install/ipfs-updater/#install-updater section, since this isn't actually an IPFS implementation, but a cmd line tool to update / install kubo

Feedback would be appreciated

**Comment by ChangingStack** [comment]
OpenBsd instructions have been confirmed to work on my computer, FreeBsd instructions should work as well. Feel free to close [https://github.com/ipfs/ipfs-docs/issues/1326](issue 1326) when/if this is merged.

**Comment by ElPaisano** [comment]
> Adding a custom TOC shouldn't be necessary. VuePress already spits one out automatically:
> 
> Screen Shot 2022-12-02 at 9 47 27 AM

Good point @johnnymatthews. Was thinking it might help to have one on the page as well, but will remove it since this would just be redundant.

Ah good callout

**Comment by ElPaisano** [comment]
@TMoMoreau @2color any feedback on this?

**Comment by TMoMoreau** [comment]
This looks good, I like the new tab system, makes the whole page look more clean.

One weird thing is that the warning callouts are greyed out, which in my mind is the opposite of what a warning callout should be. 
Screen Shot 2022-12-06 at 11 38 10 AM

File: docs/install/command-line.md [hunk 1]
```diff
@@ -1,52 +1,101 @@
...
+
+> :warning:
+> Usually, installing Kubo using one of the official binaries is the appropriate option. Building from source is only recommended if you are running Kubo on a system with severe resource constraints, or are contributing to the Kubo project.
 
 ## System requirements
 
-Kubo IPFS requires 512MiB of memory and can run an IPFS node on a Raspberry Pi. However, how much disk space your IPFS installation takes up depends on how much data you're sharing. A base installation takes up about 12MB of disk space. One can enable automatic garbage collection via [--enable-gc](/reference/kubo/cli/#ipfs-daemon) and adjust the [default maximum disk storage](https://github.com/ipfs/kubo/blob/v0.17.0/docs/config.md#datastorestoragemax) for data retrieved from other peers.
+IPFS runs on most Windows, MacOS, Linux, FreeBSD and OpenBSD systems. Ensure that your system meets the following minimum requirements:
+
+- 512MiB of memory
```
Comment (2color) [inline docs/install/command-line.md:28]: 512 or 2GB? 

(this line or the one below?)

**Comment by 2color** [comment]
Looks great! I left one comment that I found

**Comment by ElPaisano** [comment]
> This looks good, I like the new tab system, makes the whole page look more clean.
> 
> One weird thing is that the warning callouts are greyed out, which in my mind is the opposite of what a warning callout should be. Screen Shot 2022-12-06 at 11 38 10 AM

@ElPaisano look into https://docs.ipfs.tech/basics/desktop-app/#install-ipfs-desktop to match formatting

**Comment by ElPaisano** [comment]
@ElPaisano look into logos for FreeBSD and OpenBSD

File: docs/install/command-line.md (hunk 1 above)
Comment (ElPaisano) [reply docs/install/command-line.md:28]: 🤦 whoops lol @2color good catch.

I'm actually not sure. @lidel would you know the answer to this one? https://docs.ipfs.tech/install/command-line/#system-requirements says 512, https://github.com/ipfs/kubo/blob/v0.17.0/README.md#system-requirements says 2

**Comment by ElPaisano** [comment]
@TMoMoreau looks like `:::notice` isn't supported? I went with `:::warning`. Also, I removed the images for Mac, windows and Linux. see https://github.com/ipfs/ipfs-docs/pull/1342/commits/28d7184029c36e1f2f0a5deb66a4a170f335a1db

File: docs/install/command-line.md (hunk 1 above)
Comment (lidel) [reply docs/install/command-line.md]: If we mean default settings, I was unable to run it with 1GB. 
I'd say update docs to say 2GB.

**Comment by ElPaisano** [comment]
> Random thought:
> 
> looking at https://docs.ipfs.tech/install/ipfs-updater/#install-ipfs, might make sense to consolidate this into the install guide, since it's another way to install Kubo. Or even the whole https://docs.ipfs.tech/install/ipfs-updater/#install-updater section, since this isn't actually an IPFS implementation, but a cmd line tool to update / install kubo
> 
> Feedback would be appreciated

@lidel @Jorropo could I get your thoughts on this? May be out of scope for this PR, but curious what you think.

File: docs/install/command-line.md (hunk 1 above)
Comment (ElPaisano) [reply docs/install/command-line.md]: ah sorry @lidel just saw this, I was following the Slack convo

**Comment by ElPaisano** [comment]
> What should be do with the https://docs.ipfs.tech/install/command-line/#which-node-should-you-use-with-the-command-line section? As a newer user to kubo, this section still confuses me. What is it's purpose? Should it be in the install section, or should it live elsewhere? SME feedback would be helpful here.

@lidel @Jorropo would also be good to get your opinion on this. Wondering if this should go elsewhere or, at the very least, be rewritten. For example, we have info on how to configure a profile in https://docs.ipfs.tech/how-to/default-profile/ and how to configure a node in https://docs.ipfs.tech/how-to/configure-node/. So, the thought would be that info in https://docs.ipfs.tech/install/command-line/#which-node-should-you-use-with-the-command-line is actually more "advanced" info and should not be in a Kubo quickstart. Instead, we could put it elsewhere, and link to it

File: docs/install/command-line.md (hunk 1 above)
Comment (ElPaisano) [reply docs/install/command-line.md]: Slack thread https://filecoinproject.slack.com/archives/C03FFEVK30F/p1670435968403349

Comment (ElPaisano) [reply docs/install/command-line.md]: Updated in https://github.com/ipfs/ipfs-docs/pull/1342/commits/7679f3df8a11e25778805ead2af27e745b8609a1

**Comment by lidel** [comment]
:+1 for consolidating updater with command-line instructions. It is a niche tool specific to Kubo, and afaik not used by most of Kubo users (more likely, folks use docker or package from their OS, or manage it themselves).

**Comment by Jorropo** [comment]
:+1: with lidel.

**Comment by ElPaisano** [comment]
Hey all, for the idea to integrate the ipfs-update info into this page, I've opened https://github.com/ipfs/ipfs-docs/issues/1366. Main reason: I'd like to think a little more about how to best integrate that info into this page.

**Comment by ElPaisano** [comment]
With https://github.com/ipfs/ipfs-docs/pull/1342/commits/f09ce169576af392f809480667819e9f564f75e8, this should be ready to merge. @TMoMoreau can you approve?

File: docs/install/command-line.md [hunk 2]
```diff
@@ -1,52 +1,102 @@
...
+Kubo runs on most Windows, MacOS, Linux, FreeBSD and OpenBSD systems that meet the following requirements:
+
+- 6 GiB of memory.
+- 2 CPU cores (kubo is highly parallel).
+
+Note the following:
+- The amount of disk space your IPFS installation uses depends on how much data you're sharing. A base installation uses around 12MB of disk space.
+- You can enable automatic garbage collection via [--enable-gc](../reference/kubo/cli.md#ipfs-daemon) and adjust using [default maximum disk storage](https://github.com/ipfs/kubo/blob/v0.17.0/docs/config.md#datastorestoragemax) for data retrieved from other peers.
+
+### Kubo on resource-contrained systems
```
Comment (TMoMoreau) [inline docs/install/command-line.md:36]: Typo: contrained -> constrained

File: docs/install/command-line.md [hunk 3]
```diff
@@ -229,4 +402,4 @@ For example, Brave RPC API runs on port 45001, so the CLI can talk to the Brave
 
 ## Next steps
 
-Now that you've got an IPFS node installed, you can start building applications and services on top of the network! Check out the Command-line- quicks start guide and jump straight to the [Initialize the repository section](../how-to/command-line-quick-start.md#initialize-the-repository).
+Now that you've got an IPFS node installed, you can start building applications and services on top of the network! Check out the [command line quick start guide](../how-to/command-line-quick-start.md) guide.
```
Comment (TMoMoreau) [inline docs/install/command-line.md:405]: Guide is repeated here, can drop the "guide" that ends the sentence.

File: docs/install/ipfs-updater.md [hunk 4]
```diff
@@ -68,22 +77,31 @@ You can download pre-built binaries from [`dist.ipfs.tech`](https://dist.ipfs.te
    > -a----        11/5/2020   6:38 PM              0 Microsoft.PowerShell_profile.ps1
    ```
 
-7. Add the address you just copied to PowerShell's `PATH` by adding it to the end of the `Microsoft.PowerShell_profile.ps1` file stored in `Documents\WindowsPowerShell`:
+1. Add the address copied in step 5 to PowerShell's `PATH` by adding it to the end of the `Microsoft.PowerShell_profile.ps1` file stored in `Documents\WindowsPowerShell`:
 
    ```powershell
-   Add-Content C:\Users\Johnny\Documents\WindowsPowerShell\Microsoft.PowerShell_profile.ps1 "[System.Environment]::SetEnvironmentVariable('PATH',`$Env:PATH+';;C:\Users\Johnny\Apps\ipfs-update_v1.9.0\ipfs-update')"
+   Add-Content C:\Users\Johnny\Documents\WindowsPowerShell\Microsoft.PowerShell_profile.ps1 "[System.Environment]::SetEnvironmentVariable('PATH',`$Env:PATH+';;C:\Users\<username>\Apps\ipfs-update_v1.9.0\ipfs-update')"
```
Comment (TMoMoreau) [inline docs/install/ipfs-updater.md:83]: Should take the `\Johnny\` out of the first path line, change it to the variable `<username>` that is used everywhere else.

**Review by TMoMoreau — CHANGES_REQUESTED** [review]
Just some small changes requested. Looks great otherwise.

File: docs/install/command-line.md (hunk 2 above)
Comment (ElPaisano) [reply docs/install/command-line.md]: 🦅 👁️ 

Also, weird cus Vale didn't flag it when I ran it locally. I'm wondering if it was the `-`.  Anyways, weird, I'm making a note of it for the linter project

**Review by TMoMoreau — APPROVED** [review]
Looks good. Just to be clear on what's going on here, we're not consolidating the `ipfs-update` stuff into the same page just yet, right?

**Comment by ElPaisano** [comment]
> Looks good. Just to be clear on what's going on here, we're not consolidating the `ipfs-update` stuff into the same page just yet, right?

Yep, cus scope creep / I'm not sure what the best way is to do that / I'd like to completely remove the ipfs-update page from the Install section but that will require link changes so I wan't to plan a little better

**Pull request merged**